File: Segundo Trimestre/3- Ejemplos de JSON/Java/json.java
/**
 * 
 * Creación de ejemplo de control de JSON en Java.
 * 
 * @author R4f4Lc
 */

import java.net.URL;
import java.net.URLConnection;
import java.io.InputStreamReader;
import java.io.IOException;
import java.io.InputStream;

import com.google.gson.JsonElement;
import com.google.gson.JsonParser;
import com.google.gson.JsonObject;

import java.util.Scanner;

public class json {

  public static void main(String[] args) throws IOException{
    Scanner s = new Scanner(System.in);
    
    /**
     * 
     * Pedimos la IP que vamos a escanear y construimos la url a la que vamos a realizar
     * la petición.
     */ 
    System.out.println("Dime la IP a escanear: ");
    String dominio = s.nextLine();
    String sURL = "https://api.shodan.io/shodan/host/"+ dominio + "?key=p2s927StbOzkhj8BI5FOEmYQi68gTzAB";
    
    /**
     * 
     * Abrimos la conexión con shodan
     */
    URL url = new URL(sURL);
    URLConnection request = url.openConnection();
    request.connect();
    
    /**
     * 
     * Obtenemos el json de la página a la que hemos hecho la conexión.
     */
    JsonParser jp = new JsonParser();
    JsonElement root = jp.parse(new InputStreamReader((InputStream) request.getContent()));
    JsonObject rootobj = root.getAsJsonObject(); 
    
    /**
     * 
     * Mostramos por pantalla algunos de los valores del json.
     */
    System.out.println("Información de " + dominio);
    System.out.println("IP: " + rootobj.get("ip_str").getAsString());
    System.out.println("Proveedor de Servicios de Internet(ISP): " +  rootobj.get("isp").getAsString());
    System.out.println("Organización: " + rootobj.get("org").getAsString());
    System.out.println("Hostnames: " + rootobj.get("hostnames").getAsString());
    System.out.println("Puertos: " + rootobj.get("ports"));
    System.out.println("Pais: " + rootobj.get("country_name").getAsString());
    System.out.println(rootobj.get("longitude").getAsString());
    System.out.println("Longitud y Altitud: " +  rootobj.get("longitude").getAsString() + ", " + rootobj.get("latitude").getAsString());
    System.out.println("Geolocalización Google Maps: http://maps.google.com/maps?q=" + rootobj.get("latitude").getAsString() + "," + rootobj.get("longitude").getAsString());
  }
}

File: Segundo Trimestre/3- Ejemplos de JSON/Python/json.py
'''
Creación de ejemplo de control de JSON en Python.

@author: jsillero | R4f4Lc
'''
'''185.15.76.85 <-- ip ejemplo'''

print("Dime la IP a escanear:") 
ip=input()
'''Pedimos la IP que vamos a escanear y construimos la url a la que vamos a realizar la peticion'''
import json,urllib.request
'''Iniciamos conexion con shodan'''
data = urllib.request.urlopen("https://api.shodan.io/shodan/host/"+str(ip)+"?key=p2s927StbOzkhj8BI5FOEmYQi68gTzAB").read()
'''Obtenemos el json de la pagina y lo guardamos en una variable que llmamamos output'''
output = json.loads(data)

print("Información de", str(ip) )

salida = format(output.get('ip_str'))
print ('IP: ',salida)

salida = format(output.get('isp'))
print ('Proveedor de Servicios de Internet(ISP):',salida)

salida = format(output.get('org'))
print ('Organización:',salida)

salida = format(output.get('hostnames'))
print ('Hostnames: ',salida)

salida=format(output.get('ports'))
print ('Puertos: ',salida)

salida = format(output.get('country_name'))
print ('Pais: ',salida)

salida = format(output.get('longitude'))
longitud=salida
print ('Longitud',salida)

salida = format(output.get('latitude'))
latitud=salida
print ('Latitud',salida)

print("Geolocalización Google Maps: http://maps.google.com/maps?q=" +  str(latitud) + "," + str(longitud))
